### PR TITLE
Upgrade Thanos subchart v0.24 -> v0.29

### PR DESCRIPTION
**Pull request by thomasvn**
## What does this PR change?

This change primarily bumps the Thanos image version from v0.24 -> v0.29 in our existing Thanos subchart. In the process, there were some additional changes that needed to be made:

- remove the `blockSyncConcurrency` argument for the `thanos-compact` deployment since it was [deprecated in v0.27](https://thanos.io/v0.21/thanos/changelog.md/#removed-2).

It would be good to monitor this change in nightly for an extended amount of time before potentially releasing this change in Kubecost v1.99.

## Does this PR rely on any other PRs?

No

## How does this PR impact users? (This is the kind of thing that goes in release notes!)

For most users, there will be minimal impact. If they were setting up special configs for Thanos, it's possible that their configs will not be backwards compatible in this new version of Thanos.

## Links to Issues or ZD tickets this PR addresses or fixes

- https://github.com/kubecost/cost-analyzer-helm-chart/issues/1758

## How was this PR tested?

Deployed to my local Thanos cluster and monitor logs for any outlier error messages.

## Have you made an update to documentation?

Not yet. However I don't believe this requires an addition to https://docs.kubecost.com.

### CODE REVIEW

File: cost-analyzer/Chart.yaml
Comment (LaikaN57) [inline cost-analyzer/Chart.yaml:6]: Please update to `1.98.0` as this is the current version of the `develop` branch.

**Review by LaikaN57 — APPROVED** [review]
LGTM

File: cost-analyzer/Chart.yaml
Comment (michaelmdresser) [inline cost-analyzer/Chart.yaml]: Would you omit the appVersion update? I think we'll hit a merge conflict between this and https://github.com/kubecost/cost-analyzer-helm-chart/pull/1793. https://github.com/kubecost/cost-analyzer-helm-chart/pull/1793 is more tightly scoped to the appVersion update so I'd prefer to do it there.

**Review by michaelmdresser — APPROVED** [review]

File: cost-analyzer/Chart.yaml
Comment (thomasvn) [reply cost-analyzer/Chart.yaml]: Good point, will do!

**Comment by thomasvn** [comment]
Before merging, I'd like to upgrade the Thanos image version from v0.28.1 to v0.29.0 which was just released a few days ago.

**Comment by AjayTripathy** [comment]
Approved, let's merge then make sure we get this tested on the scale cluster. Might be a good time to transition/knowledge share this with you @Sean-Holcomb can you help Thomas get set up with it?

**Review by AjayTripathy — APPROVED** [review]

**Comment by thomasvn** [comment]
Holding off on merging this PR since we've discovered that this change would not actually fix Issue https://github.com/kubecost/cost-analyzer-helm-chart/issues/1758

**Comment by AjayTripathy** [comment]
May as well do this upgrade since the works been done even if its not a fix for 1758

**Pull request merged**
